### PR TITLE
Fix race conditions on Init

### PR DESCRIPTION
**Pull request by gpotter2**
fixes https://github.com/secdev/scapy/issues/1966

### CODE REVIEW

File: scapy/packet.py [hunk 1]
```diff
@@ -219,32 +220,38 @@ def prepare_cached_fields(self, flist):
...
         # Fields initialization
         for f in flist:
             if isinstance(f, MultipleTypeField):
-                del Packet.class_default_fields[cls_name]
-                del Packet.class_default_fields_ref[cls_name]
-                del Packet.class_fieldtype[cls_name]
-                del Packet.class_packetfields[cls_name]
+                # Abort
                 self.class_dont_cache[cls_name] = True
                 self.do_init_fields(self.fields_desc)
```
Comment (guedou) [inline scapy/packet.py]: I think that this method could be called twice in a racing scenario. If https://github.com/secdev/scapy/blob/master/scapy/packet.py#L185 is called twice, the resulting packet will be broken.

**Comment by guedou** [comment]
That is not my definition of a simple fix =)

I am ok with the PR if it fixes the initial issue, however I still think that a race condition can appear.

**Comment by guedou** [comment]
@gpotter2 could you rebase?

**Comment by gpotter2** [comment]
Do you know if this actually fixes the issue ?

**Comment by guedou** [comment]
I am able to quickly (less than 50 runs of the PoC with n set to 2000) reproduce the issue on master on two different setups. 1000 runs of the same PoC with this branch did not trigger the issue.

**Comment by gpotter2** [comment]
Right. `MultipleTypeField` doesn't like the new system. It should be possible to work around that though

**Comment by gpotter2** [comment]
@guedou
Fixed !

**Pull request merged**